src/components/application_manager/rpc_plugins/vehicle_info_plugin/src/pending_resumption_handler.cc

```diff
@@ -0,0 +1,229 @@
+  }
+
+  if (succesfull_response) {
+    for (auto key : request_keys) {
+      if (!Contains(key, response_keys)) {
+        subscription_results[key] = true;
+      } else {
+        const auto kSuccess =
+            hmi_apis::Common_VehicleDataResultCode::VDRC_SUCCESS;
+        const auto vd_result_code =
+            response_params[key][application_manager::strings::result_code]
+                .asInt();
+        subscription_results[key] = vd_result_code == kSuccess;
+      }
+    }
+  }
+  return subscription_results;
+}
+
+void PendingResumptionHandler::HandleResumptionSubscriptionRequest(
+    VehicleInfoAppExtension& extension,
+    resumption::Subscriber subscriber,
+    application_manager::Application& app) {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  std::set<std::string> subscriptions = GetExtensionSubscriptions(extension);
+
+  smart_objects::SmartObjectSPtr request =
+      CreateSubscribeNotificationToHMI(subscriptions);
+
+  smart_objects::SmartObject& request_ref = *request;
+  const auto function_id = static_cast<hmi_apis::FunctionID::eType>(
+      request_ref[application_manager::strings::params]
+                 [application_manager::strings::function_id].asInt());
+  const uint32_t corr_id =
+      request_ref[application_manager::strings::params]
+                 [application_manager::strings::correlation_id].asUInt();
+
+  auto resumption_request =
+      MakeResumptionRequest(corr_id, function_id, *request);
+
+  subscribe_on_event(function_id, corr_id);
+
+  if (pending_requests_.empty()) {
+    LOG4CXX_DEBUG(logger_, "There are no pending requests");
+    pending_requests_[corr_id] = request_ref;
+    subscriber(app.app_id(), resumption_request);
+    application_manager_.GetRPCService().ManageHMICommand(request);
+    return;
+  } else {
+    LOG4CXX_DEBUG(logger_, "There are pending requests");
+    FrozenResumption frozen_res(app.app_id(), extension, subscriber);
+    freezed_resumptions_.push(frozen_res);
+  }
+}
+
+resumption::ResumptionRequest PendingResumptionHandler::MakeResumptionRequest(
+    const uint32_t corr_id,
+    const hmi_apis::FunctionID::eType function_id,
+    const smart_objects::SmartObject& message) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  resumption::ResumptionRequest resumption_request;
+  resumption_request.request_ids.correlation_id = corr_id;
+  resumption_request.request_ids.function_id = function_id;
+  resumption_request.message = message;
+  return resumption_request;
+}
+
+std::set<std::string> PendingResumptionHandler::GetExtensionSubscriptions(
+    VehicleInfoAppExtension& extension) {
+  std::set<std::string> subscriptions;
+  for (auto& ivi : application_manager::MessageHelper::vehicle_data()) {
+    const auto it = extension.Subscriptions().find(ivi.second);
+    if (extension.Subscriptions().end() != it) {
+      subscriptions.insert(ivi.first);
+    }
+  }
+  return subscriptions;
+}
+
+smart_objects::SmartObjectSPtr
+PendingResumptionHandler::CreateSubscribeNotificationToHMI(
```

Comment:
@mked-luxoft You are sending request to hmi. NOT notification...

Reply:
@AByzhynar fixed in 23e9b6736a69f55e45d495d644b00c6437b37002